protocol/tx_test.go

```diff
@@ -1,4 +1,4 @@
-package protocol
```

Comment:
I moved `Snapshot` into `protocol` based on the observation that the `Chain` and `Snapshot` objects together embody the blockchain state described by the spec; neither one does it alone. It was a small part of much larger streamlining.

However, I was easily able to [back out that one piece of it](https://github.com/chain/chain/commit/3872e6ede101fc0c38fffd2cbd2e6226141638c1) on the `entries-validation` branch and it does in fact clarify the dependencies, so I'll repeat that here momentarily.